react/src/components/timeline-app/Application.jsx

```diff
@@ -0,0 +1,45 @@
+import MockDataService from './MockDataService';
+import React from 'react';
+import TimelineComponent from './TimelineComponent';
+
+
+class Application extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      data: null,
+    };
+  }
+
+
+  componentWillMount() {
+    MockDataService.fetchData(null).then((data) => {
+      this.setState({ data });
+    });
+  }
+
+
+  render() {
+    if (this.state.data) {
+      return (
+        <div>
+          <h4>Chronicles of { this.state.data.name }</h4>
+          <TimelineComponent data={ this.state.data }/>
+        </div>
+      );
+    }
+    return (
+      <div>
+        Loading data...
+      </div>
+    );
+  }
```

Comment:
also just for info: to avoid 2 returns, you could use either ternary:
```js
return ( 
  <div>
   this.state.data 
       ? // render divs here...
       : // render other divs here...
 </div>
)

or even 

this.state.data && ( 
  // render divs here
)

!this.state.data && (
 // render other divs here
)

&& would implicitly convert this.state.data to boolean (false / true), react would then evaluate them and based on them will display the expression in the brackets w/o outputting false, true 
```

https://facebook.github.io/react/docs/conditional-rendering.html#inline-if-with-logical--operator